### PR TITLE
Run integration tests on travis

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #185 

#### What's this PR do?
Runs integration tests on travis. It sets up the minimal parts of edX and then runs the tests within the context of that edX docker setup

#### How should this be manually tested?
Tests should pass

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+    - install:
+      - pip install tox
+      script: tox -e django18
+      env:
+        name: Django 1.8
+    - install:
+      - pip install tox
+      script: tox -e django111
+      env:
+        name: Django 1.11
```
Comment (gsidebo) [inline .travis.yml]: Can you name these `Django 1.8 Unit Tests` and `Django 1.11 Unit Tests` or something like that?

File: run_sga_tests.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
```
Comment (gsidebo) [inline run_sga_tests.sh]: `run_devstack_integration_tests.sh` would be a better name for this file IMO

File: .travis.yml [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
-env:
-    - TOX_ENV=django18
-    - TOX_ENV=django111
-script:
-    - tox -e $TOX_ENV
+matrix:
+  include:
+    - install:
+      - git clone https://github.com/edx/devstack.git
+      - (cd devstack; sed -i 's/:cached//g' ./docker-compose-host.yml)
```
Comment (gsidebo) [inline .travis.yml]: Can you include a comment explaining this and why it's necessary to replace ":cached" in the file?

File: .travis.yml [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
+      - (cd devstack; make dev.clone)
+      script:
+        - (cd devstack; DEVSTACK_WORKSPACE=$PWD/.. docker-compose -f docker-compose.yml -f docker-compose-host.yml run -v $PWD/..:/edx-sga lms /edx-sga/run_sga_tests.sh)
```
Comment (gsidebo) [inline .travis.yml]: I think I understand why you're mounting that directory to `/edx-sga` (so you know where it is within the int. test shell script), but does it not make more sense to copy the working directory to `/edx-sga` as the first step in `install`? (1) This kind of obscures the fact that we're putting the edx-sga repo at that location, and (2) by the time you're doing this mount, that directory has all of the cloned edx repos in it as well, right? I don't think we need those. Might be misreading this, though

File: run_sga_tests.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# the next build will not need to install them
+# from scratch again.
+pip install --exists-action w -r requirements/edx/pre.txt
+pip install --exists-action w -r requirements/edx/github.txt
+pip install --exists-action w -r requirements/edx/local.txt
+
+# HACK: within base.txt stevedore had a
+# dependency on a version range of pbr.
+# Install a version which falls within that range.
+pip install  --exists-action w pbr==0.9.0
```
Comment (gsidebo) [inline run_sga_tests.sh]: Nit: 2x white space after `install`

File: run_sga_tests.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+pip install --exists-action w -r requirements/edx/django.txt
+pip install --exists-action w -r requirements/edx/base.txt
+pip install --exists-action w -r requirements/edx/paver.txt
+pip install --exists-action w -r requirements/edx/testing.txt
+if [ -e requirements/edx/post.txt ]; then pip install --exists-action w -r requirements/edx/post.txt ; fi
+
+
+cd /edx-sga
+pip uninstall edx-sga -y
```
Comment (gsidebo) [inline run_sga_tests.sh]: Nit: To me it's clearer if we do this uninstall right before we copy the directory to lms/djangoapps. It's more obvious that we're replacing any currently-installed version of the package with the repo as it is

**Comment by noisecapella** [comment]
Looks like the tests were failing because of different versions between `test_requirements.txt` and what was on edX. I've changed the script to not use `test_requirements.txt` and just leave that for the unit tests, all test packages should already exist on edx anyway

**Pull request merged**